app/controllers/benefits/CompanyBenefitsSummaryController.scala

```diff
@@ -0,0 +1,99 @@
+import com.google.inject.name.Named
+import controllers.{ErrorPagesHandler, TaiBaseController}
+import controllers.auth.AuthJourney
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.tai.config.ApplicationConfig
+import uk.gov.hmrc.tai.model.TaxYear
+import uk.gov.hmrc.tai.model.domain.TemporarilyUnavailable
+import uk.gov.hmrc.tai.service.benefits.BenefitsService
+import uk.gov.hmrc.tai.service.journeyCache.JourneyCacheService
+import uk.gov.hmrc.tai.service.journeyCompletion.EstimatedPayJourneyCompletionService
+import uk.gov.hmrc.tai.service.{EmploymentService, TaxAccountService}
+import uk.gov.hmrc.tai.util.constants.TaiConstants.UpdateIncomeConfirmedAmountKey
+import uk.gov.hmrc.tai.viewModels.IncomeSourceSummaryViewModel
+import views.html.benefits.CompanyBenefitsView
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+import scala.util.control.NonFatal
+
+class CompanyBenefitsSummaryController @Inject() (
+  val auditConnector: AuditConnector,
+  @Named("Update Income") journeyCacheService: JourneyCacheService,
+  taxAccountService: TaxAccountService,
+  employmentService: EmploymentService,
+  benefitsService: BenefitsService,
+  estimatedPayJourneyCompletionService: EstimatedPayJourneyCompletionService,
+  authenticate: AuthJourney,
+  applicationConfig: ApplicationConfig,
+  mcc: MessagesControllerComponents,
+  companyBenefits: CompanyBenefitsView,
+  implicit val errorPagesHandler: ErrorPagesHandler
+)(implicit ec: ExecutionContext)
+    extends TaiBaseController(mcc) {
+
+  def onPageLoad(empId: Int): Action[AnyContent] = authenticate.authWithValidatePerson.async { implicit request =>
+    val nino = request.taiUser.nino
+
+    val cacheUpdatedIncomeAmountFuture =
+      journeyCacheService.currentValueAsInt(s"$UpdateIncomeConfirmedAmountKey-$empId")
+
+    (
+      taxAccountService.taxCodeIncomes(nino, TaxYear()),
+      employmentService.employment(nino, empId),
+      benefitsService.benefits(nino, TaxYear().year),
+      estimatedPayJourneyCompletionService.hasJourneyCompleted(empId.toString),
+      cacheUpdatedIncomeAmountFuture
+    ).mapN {
+      case (
+            Right(taxCodeIncomes),
+            Some(employment),
+            benefitsDetails,
+            estimatedPayCompletion,
+            cacheUpdatedIncomeAmount
+          ) =>
+        val rtiAvailable = employment.latestAnnualAccount.exists(_.realTimeStatus != TemporarilyUnavailable)
+
+        val incomeDetailsViewModel = IncomeSourceSummaryViewModel(
+          empId,
+          request.fullName,
+          taxCodeIncomes,
+          employment,
+          benefitsDetails,
+          estimatedPayCompletion,
+          rtiAvailable,
+          applicationConfig,
+          cacheUpdatedIncomeAmount
+        )
+
+        if (!incomeDetailsViewModel.isUpdateInProgress) {
+          journeyCacheService.flushWithEmpId(empId)
```

Comment:
Are we OK to leave this running in background or should it be mapped onto following statement?